lib/facter/resolvers/augeas_resolver.rb

```diff
@@ -20,7 +20,13 @@ def read_augeas_version(fact_name)
         end
 
         def read_augeas_from_cli
-          output = Facter::Core::Execution.execute('augparse --version 2>&1', logger: log)
+          command = if RbConfig::CONFIG['host_os'] =~ /mswin|msys|mingw|cygwin|bccwin|wince|emc/
+                      'augparse'
+                    else
+                      '/opt/puppetlabs/puppet/bin/augparse'
```

Comment:
Do we want to have this fact only in a puppet-agent context? What if i want to use the augeas fact with standalone facter and i install `augeas` to have `augparse` command?

For example installing augeas-tools and cfacter 3.11.0 on ubuntu-20.04 i can have the augeas fact.

Reply:
I change the behaviour as follows:
1st: check if the path from puppet-agent's `augparse` is exists and is readable; If not
2st: call 'augparse' without path and hope it is in path; If this command returns empty string
3rd: try requiring augeas gem and calling Augeas cli method to determine it's version.

So if augeas is present on the system, facter will determine it's version, even if it isn't in path or if it was installed manually and not shipped in puppet-agent context.